### ISSUE TITLE
CellEditor: Multiple 'rendered' editors are ignored

### ISSUE THREAD

**Issue by mdzaebel** [primefaces/primefaces#3956]
## 1) Environment
PF versions: All, AppServer: WildFly 11 / JEE7, All browsers
## 2) Expected behavior
Input for the first cellEditor should appear also in the first column
## 3) Actual behavior
Output is correctly displayed regarding 'rendered', while the input editor only appears clicking the second column. Only the last cellEditor is used for inputs. Same happens for _p:treeTable_ but there, the first input appears as 'in1' from the second cellEditor, so once again, the last cellEditor wins.

## 5) Sample XHTML
``` html
<html xmlns="http://www.w3.org/1999/xhtml" xmlns:h="http://java.sun.com/jsf/html"
      xmlns:f="http://xmlns.jcp.org/jsf/core" xmlns:p="http://primefaces.org/ui">
<h:head/><h:body>
  <p:dataTable value="#{[0]}" var="v" editMode="cell" editable="true" tableStyle="width:auto">
    <p:columns var="c" value="#{[0,1]}">
      <p:cellEditor rendered="#{c==0}">
        <f:facet name="input">in0</f:facet>
	<f:facet name="output">out0</f:facet>
      </p:cellEditor>
      <p:cellEditor rendered="#{c==1}">
        <f:facet name="input">in1</f:facet>
        <f:facet name="output">out1</f:facet>
      </p:cellEditor>
    </p:columns>
  </p:dataTable>
</h:body></html>
```
![image](https://user-images.githubusercontent.com/415392/43920896-77906614-9c1a-11e8-8ccb-02e2437b17e0.png)      The output emerges - as expected - from both p:cellEditor elements
![image](https://user-images.githubusercontent.com/415392/43920976-b5ab001c-9c1a-11e8-803f-05345c995d5b.png)       The input editor only appears by clicking on the _second_ column

**Comment by tandraschko** [comment]
A PR is very welcome.

**Comment by mdzaebel** [comment]
If you want to say, that this behavior is expected (not a defect), than it should be documented, what kind of constructs are allowed. This would save us all a lot of time.

**Comment by tandraschko** [comment]
Currently expected as it was never implemented to support that case. So it would be a enhancement of the current Implementation.

**Comment by mdzaebel** [comment]
Well than, this issue should be handled as a documentation bug. If there are non standard restrictions in the allowed markup for certain components, it would be of great value to have a note in the documentation. How should your client be able to anticipate these restrictions? The natural result is, that many users try (otherwise recommended) constructs thinking that PF has a bug or they don't understand. Both is extremely time consuming or leads to using other technologies! To reduce the problem of this issue cost me many hours and you are also involved. Note that many users will not be able to dig into such problems, especially when they are hidden in greater markup sections. So the usability of PF increases dramatically, if common pitfalls are excluded via, say, a short restriction note (if necessary). Every PF component should be preventively examined about this problem. Languages usually have 'grammars' but it might be enough to concentrate on most important restrictions.

Note that the documentation only mention p:dataTable but in you showcases, p:cellEditor is also used in p:treeTable. 

In short, it must be clear **where** and **how** a component can be used.

**Comment by kukel** [comment]
@mdzaebel: I sort of agree, but there has never been an attempt to test all permutations of components in components or even just features of components. The most common use cases are sort of automatically tested when features are implemented (and hence most common pitfalls prevented, emphasis on common, since this feature has been there for several years and you are the first person to report it) . So although it might be great to have an extensive (fully complete!) list, it is sort of undoable.  Oh and for your information, other frameworks have the same 'problems', believe me, I tried several.

**Comment by mdzaebel** [comment]
Good to know! That's because I said, "it might be enough to concentrate on most important restrictions". I would recommend just to consider "**important** restrictions" for each component, as it's sometimes really difficult to find the problem within a more complex structures.

It might be best to implement Primefaces in a way, thas minimizes surprizes of this kind.

My workaround here is, to write the logic into one cellEditor, as I now know the restriction. Hope that PF respects that "it must be clear **where** and **how** a component can be used", even if you need multiple iterations and can't be complete.

**Comment by melloware** [comment]
Closing this as wontfix.  Years open and no one else raising this corner case
